### PR TITLE
tock-cells/OptionalCell: rename extract-method to get

### PR DESCRIPTION
**Pull request by lschuermann**
### Pull Request Overview

In response to https://github.com/tock/tock/pull/2531#issuecomment-1625703637, this renames `OptionalCell::extract` to `OptionalCell::get`. Hence this reverts commit 3136f00a77173708625256011be54ee24778364f.

I agree that `extract` is extremely confusing in this context -- we're operating on a cell, and `extract` could reasonably suggest that it removes the value from the `OptionalCell`. This is amplified by the fact that `take` (the natural method name for a "destructive get" in `TakeCell`) implies move-semantics, which we don't have in `OptionalCell<T>`, as it requires that `T: Copy`. Even though I originally made the `get -> extract` change, I still get this wrong.

Now, do I want to open this can of worms again? I don't know. I still believe we need a method on `OptionalCell` whose signature is `fn somename(&self) -> Option<T>`. Is `get` the right name? At least some evidence (in @kupiakos) seems to suggest that it's not entirely atypical for a `get` method to return an `Option<T>`, and it is the only interpretation of `get` that makes sense given `OptionalCell`'s semantics. On the other hand, I think the concerns over this name as raised on the initial PR (#2531) are fair.

### Testing Strategy

This pull request was tested by compiling.

### TODO or Help Wanted

This pull request still needs bike-shedding of the method name.

### Documentation Updated

- [x] ~Updated the relevant files in `/docs`, or~ no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Comment by kupiakos** [comment]
> the concerns over this name as raised on the initial PR (https://github.com/tock/tock/pull/2531) are fair

For reference, I've [replied](https://github.com/tock/tock/pull/2531#issuecomment-1625703637) to those concerns on that PR.

> `take` (the natural method name for a "destructive get" in `TakeCell`) implies move-semantics, which we don't have in `OptionalCell<T>`, as it requires that `T: Copy`

[`OptionalCell<T>::take`](https://github.com/tock/tock/blob/ee3973f/libraries/tock-cells/src/optional_cell.rs#L140-L142) is already present and does not require `T: Copy`. It doesn't need to - `take` semantics don't requiring cloning, just like how [`core::cell::Cell<T>::take`](https://doc.rust-lang.org/std/cell/struct.Cell.html#method.take) doesn't require `T: Copy` as its [`impl` block only requires `T: Default`](https://doc.rust-lang.org/std/cell/struct.Cell.html#impl-Cell%3CT%3E-3)

**Review by hudson-ayers — APPROVED** [review]
I am on board with this based on the discussion on the call

**Review by bradjc — APPROVED** [review]
At least personally what I always do is 1. call .get() 2. compiler error 3. google optionalcell 4. look for the function that has the correct signature 5. oh right call extract(). So, I guess I'm in favor of reverting.

File: libraries/tock-cells/src/optional_cell.rs [hunk 1]
```diff
@@ -154,8 +154,8 @@ impl<T> OptionalCell<T> {
 }
 
 impl<T: Copy> OptionalCell<T> {
-    /// Returns a copy of the contained [`Option`].
-    pub fn extract(&self) -> Option<T> {
+    /// Returns a copy the contained [`Option`].
+    pub fn get(&self) -> Option<T> {
```
[the remark this answers is not in the document]
Comment (lschuermann) [reply libraries/tock-cells/src/optional_cell.rs:158]: Sounds good, will add!

File: libraries/tock-cells/src/optional_cell.rs [hunk 2]
```diff
@@ -155,6 +155,34 @@ impl<T> OptionalCell<T> {
 
 impl<T: Copy> OptionalCell<T> {
     /// Returns a copy the contained [`Option`].
```
Comment (brghena) [inline libraries/tock-cells/src/optional_cell.rs:157]:
```suggestion
    /// Returns a copy of the contained [`Option`].
```

**Review by brghena — APPROVED** [review]

File: libraries/tock-cells/src/optional_cell.rs (hunk 2 above)
Comment (lschuermann) [reply libraries/tock-cells/src/optional_cell.rs]: Thanks!

**Review by brghena — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]

**Pull request merged**